flocker/node/test/test_script.py

```diff
@@ -915,3 +920,46 @@ class ContainerAgentOptionsTests(
     """
     Tests for ``ContainerAgentOptions``.
     """
+
+
+class GetExternalIPTests(SynchronousTestCase):
+    """
+    Tests for ``_get_external_ip``.
+    """
+    def setUp(self):
+        server = socket.socket()
+        server.bind(('127.0.0.1', 0))
+        server.listen(5)
+        self.destination_port = server.getsockname()[1]
+        self.addCleanup(server.close)
+
+    @capture_logging(lambda test, logger:
+                     assertHasAction(test, logger, LOG_GET_EXTERNAL_IP, True,
+                                     {u"host": u"localhost",
+                                      u"port": test.destination_port},
+                                     {u"local_ip": u"127.0.0.1"}))
+    def test_successful_get_external_ip(self, logger):
+        """
+        A successful external IP lookup returns the local interface's IP.
+        """
+        self.assertEqual(u"127.0.0.1",
+                         _get_external_ip(u"localhost", self.destination_port))
```

Comment:
Makes some assumptions about the system's network stack/configuration. :/  Really hard to avoid doing that without just faking the `socket` module, though ... :/

Still, I'm not looking forward to when we have to start ignoring failures from this test due to stupid network behaviors/misbehaviors.  So maybe faking the socket module is actually the better course?  Hard to decide which way is less bad.  I think fake socket might be since the fake can eventually be verified and then all the gross real socket/system interaction goes in one well-defined place instead of spread across the rest of the project.